### PR TITLE
Support to enable rocksdbs statistics

### PR DESCRIPTION
**Pull request by chenxu14**
I think there is a need to enable rocksdb's statistics, so we can dump some metrics to the log, and use it to find bottleneck.
We can enable this by specify --rocksdb_statistics_enable=true in nebula-storaged.conf, and declare the stats level through --rocksdb_stats_level(default value is kExceptHistogramOrTimers)

### CODE REVIEW

**Comment by Shylock-Hg** [comment]
If so I think you could export the option `stats_dump_period_sec` too.

File: src/kvstore/RocksEngineConfig.cpp [hunk 1]
```diff
@@ -186,5 +194,27 @@ bool loadOptionsMap(std::unordered_map<std::string, std::string> &map, const std
...
+    } else if (str == "kExceptTimeForMutex") {
+        return rocksdb::StatsLevel::kExceptTimeForMutex;
+    } else {
+        return rocksdb::StatsLevel::kAll;
+    }
+}
+
+std::shared_ptr<rocksdb::Statistics> getDBStatistics() {
+    if (FLAGS_rocksdb_statistics_enable) {
+        static std::shared_ptr<rocksdb::Statistics> dbstats = rocksdb::CreateDBStatistics();
```
Comment (Shylock-Hg) [inline src/kvstore/RocksEngineConfig.cpp:213]: Let rocksdb manage it's lifetime.

**Comment by chenxu14** [comment]
> If so I think you could export the option `stats_dump_period_sec` too.

I think we can specify this in --rocksdb_db_options

File: src/kvstore/RocksEngineConfig.cpp (hunk 1 above)
Comment (chenxu14) [reply src/kvstore/RocksEngineConfig.cpp:213]: We should pass the same Statistics object multiple DBs, in oder to aggregated values for all those DBs. so it's better to manage it ourselfs?

Comment (Shylock-Hg) [reply src/kvstore/RocksEngineConfig.cpp:213]: If you share this cross db instances, the statistics may be mixed. The log of one db may contains other db's metrics data.

Comment (chenxu14) [reply src/kvstore/RocksEngineConfig.cpp:213]: Exactly true, IMHO dump to log is not the perfect way, we want report those metrics to the monitor system directly (such as Falcon), If do so, getDBStatistics is needed.
Can we disable rocksdb's dump feature and manage it ourself, I can do this in another PR.

Comment (Shylock-Hg) [reply src/kvstore/RocksEngineConfig.cpp:213]: Even if so I suggest you manage the one DBStatistics object to one DB instance.

Comment (chenxu14) [reply src/kvstore/RocksEngineConfig.cpp:213]: You mean we'd better monitor each DB separately?

Comment (Shylock-Hg) [reply src/kvstore/RocksEngineConfig.cpp:213]: Yes, I thinks so.

Comment (chenxu14) [reply src/kvstore/RocksEngineConfig.cpp:213]: I think there is no need, since they belong to the same space,  just different partitions only.

Comment (Shylock-Hg) [reply src/kvstore/RocksEngineConfig.cpp:213]: No, there will be multiple DB instance in the storage process.

Comment (critical27) [reply src/kvstore/RocksEngineConfig.cpp:213]: Em, since we shared block_cache among all instance, so maybe we could collect stats of all instance.

Comment (Shylock-Hg) [reply src/kvstore/RocksEngineConfig.cpp:213]: Ok, agree now. We could collect the all storage Statistics in one currently.

File: src/kvstore/RocksEngineConfig.cpp [hunk 2]
```diff
@@ -186,5 +194,27 @@ bool loadOptionsMap(std::unordered_map<std::string, std::string> &map, const std
...
+        return dbstats;
+    }
+    return nullptr;
+}
+
```
Comment (dutor) [inline src/kvstore/RocksEngineConfig.cpp:218]: Suggest to put the creation and initialization of `rocksdb::Statistics`  into a single function.

File: src/kvstore/RocksEngineConfig.cpp [hunk 3]
```diff
@@ -60,6 +60,9 @@ DEFINE_string(rocksdb_compression_per_level, "", "Specify per level compression
                                                  "e.g. \"no:no:lz4:lz4::zstd\" === "
                                                  "\"no:no:lz4:lz4:lz4:snappy:zstd:snappy\"");
 
+DEFINE_bool(rocksdb_statistics_enable, false, "Whether to export rocksdb's statistics");
```
Comment (dutor) [inline src/kvstore/RocksEngineConfig.cpp:63]: It would be better to name this flag to conform to others, like `--enable_partitioned_index_filter`

**Comment by chenxu14** [comment]
Thanks for your advise, can you help to review again.@dutor

**Comment by dangleptr** [comment]
Please check your code style with cpplint.
Thanks @chenxu14

**Comment by chenxu14** [comment]
> Please check your code style with cpplint.
> Thanks @chenxu14

OK, fixed

**Pull request merged**